tuna/miopen/celery_tuning/tuning.py

```diff
@@ -424,3 +398,51 @@ def drain_queue(worker_type, dbt, event):
         continue
 
   return True
+
+async def consume(job_counter, worker_type, dbt):
+  """Retrieve celery results from redis db"""
+  redis = await aioredis.from_url(
+      f"redis://{TUNA_CELERY_BROKER}:{TUNA_REDIS_PORT}/15")
+  LOGGER.info('Connected to redis')
+  while job_counter.value > 0:
+    cursor = "0"
+    keys = []
+    while cursor != 0:
+      cursor, results = await redis.scan(cursor, match='*'
+                                        )  # update with the celery pattern
+      keys.extend(results)
+    LOGGER.info('Found results')
+    for key in keys:
+      LOGGER.info(type(key))
+      try:
+        data = await redis.get(key)
+        if data:
+          await result_callback(key.decode('utf-8'), data.decode('utf-8'), worker_type, dbt)
```

Comment:
You can have your callback work on batches of data items, to make data insertion in the database more effective.

Reply:
we dont have that support on the parsing side yet. But yes that is the plan